src/metabase/pulse/render/datetime.clj

```diff
@@ -11,33 +11,107 @@
            java.time.Period
            java.time.temporal.Temporal))
 
+(defn temporal-string?
+  "Returns `true` if the string `s` is parseable as a datetime.
+
+  `(temporal-string? \"asdf\")` -> false
+  `(temporal-string? \"2020-02-02\")` -> true"
+  [s]
+  (boolean
+    (try
+      (u.date/parse s)
+      (catch Exception _e false))))
+
 (defn- reformat-temporal-str [timezone-id s new-format-string]
   (t/format new-format-string (u.date/parse s timezone-id)))
 
+(defn- day-of-week
+  [s abbreviate]
+  (get-in
+    {"1" {:full (tru "Sunday") :abbreviated (tru "Sun")}
+     "2" {:full (tru "Monday") :abbreviated (tru "Mon")}
+     "3" {:full (tru "Tuesday") :abbreviated (tru "Tue")}
+     "4" {:full (tru "Wednesday") :abbreviated (tru "Wed")}
+     "5" {:full (tru "Thursday") :abbreviated (tru "Thu")}
+     "6" {:full (tru "Friday") :abbreviated (tru "Fri")}
+     "7" {:full (tru "Saturday") :abbreviated (tru "Sat")}}
+    (if abbreviate
+      [s :abbreviated]
+      [s :full])))
+
+(defn- month-of-year
+  [s abbreviate]
```

Comment:
I think this could better be done as 
```clojure
datetime=> (let [fmtr (java.time.format.DateTimeFormatter/ofPattern "MMMM")]
             (.format fmtr (java.time.Month/of 2)))
"February"
datetime=> (let [fmtr (java.time.format.DateTimeFormatter/ofPattern "MMM")]
             (.format fmtr (java.time.Month/of 2)))
"Feb"
datetime=> (java.util.Locale/setDefault java.util.Locale/FRENCH)
nil
datetime=> (let [fmtr (java.time.format.DateTimeFormatter/ofPattern "MMMM")]
             (.format fmtr (java.time.Month/of 2)))
"février"
datetime=> (let [fmtr (java.time.format.DateTimeFormatter/ofPattern "MMM")]
             (.format fmtr (java.time.Month/of 2)))
"févr."
```